src/util/argParser.ts

```diff
@@ -714,6 +725,23 @@ class ArgParser {
       }
     }
 
+    let selectLinks: ExtractSelector[];
+
+    if (argv.selectLinks) {
+      selectLinks = argv.selectLinks.map((x: string) => {
+        const parts = x.split("->");
+        const selector = parts[0];
+        const value = parts[1] || "";
+        const extract = parts.length > 1 ? value.replace("@", "") : "href";
+        const isAttribute = value.startsWith("@");
+        return { selector, extract, isAttribute };
+      });
+    } else {
+      selectLinks = DEFAULT_SELECTORS;
+    }
```

Comment:
If the syntax for a link selector passed in is broken, the failure is a bit silent - i.e. the crawler obviously won't find other pages, but it does finish with a 0 exit code and no error messages. I think that's not the most user-friendly behavior here. At minimum I'd suggest we log an error message, but I'm wondering if we should just fail the crawl early with a fatal message given how important link extraction is to almost all crawl scope types.

Reply:
Added validation with https://www.npmjs.com/package/css-selector-parser, will fail instantly on invalid selectors.